src/odemis/driver/actuator.py

```diff
@@ -1071,21 +1055,33 @@ def moveAbs(self, pos):
 
         return f
 
-    def _doMoveAbs(self, pos):
-        axis, distance = pos.items()[0]
-
+    def _referenceIfNeeded(self):
+        """
+        Force a referencing if it has moved a certain number of times
+        """
         # Reference from time to time
         self._move_num += 1
         if self._ref_period and self._move_num >= self._ref_period:
             # Axis might always go towards negative direction during referencing. Make
             # sure that the referencing works properly in case of a negative position (especially
             # important if the actuator is cyclic).
             logging.debug("Moving to reference starting position %s", self._ref_start)
-            self._doMoveAbs({self._axis: self._ref_start})
-            logging.debug("Referencing axis %s (-> %s) after %s moves." % self._axis, self._caxis, self._ref_period)
+            self._child.moveAbsSync({self._caxis: self._ref_start + self._offset})
+            logging.debug("Referencing axis %s (-> %s) after %s moves", self._axis, self._caxis, self._move_num)
             self._child.reference({self._caxis}).result()
             self._move_num = 0
```

Comment:
Maybe it's safer to reset the counter before moveabssync. Then we should also be able to call domoveabs, because the next time domoveabs is called, the if block will be ignored and we don't get the recursion problem.

Now it seems that the executor is making sure that this doesn't happen, so that also works.